File: problem_15/problem_15.c
#include <stdio.h>

int nCr(int n,int r);
int factorial(int n);

int main() {
  int n;
  printf("Please enter a number here:\n");
  scanf("%d",&n);
  printf("Here is your answer: %d\n",nCr(2*n,n));
  return 0;
}

int nCr(int n,int r) {
  int result = factorial(n) / (factorial(r) * factorial(n-r));
  return result;
}

int factorial(int n) {
  int result = 1;
  while (n > 1) {
    result *= n;
    n--;
  }
  return result;
}

File: problem_15/problem_15.js
const euler_15 = n => nCr(2*n,n);

const nCr = (n,r) => {
  if (r > n || n < 1 || r < 1) {
    return undefined;
  }

  return factorial(n) / (factorial(r) * factorial(n-r));
};

const factorial = num => {
  let result = 1,
      n = +num;
  while (n > 1) {
    n--;
    result *= n;
  }
  return result;
}
